src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/LibraryLookup.java

```diff
@@ -110,17 +111,14 @@ static LibraryLookup ofDefault() {
      * @param path the library path.
      * @return a library lookup object for given path.
      */
-    static LibraryLookup ofPath(String path) {
+    static LibraryLookup ofPath(Path path) {
         Objects.requireNonNull(path);
+        String absolutePath = path.toAbsolutePath().toString();
```

Comment:
Path.toAbsolutePath itself is a secure operation (could potentially throw SecurityException).  Should insist on absolute Path (that way it is up to the client to manage absolute Path object)?

Reply:
I think it's nice, from an usability perspective, that any path is accepted - but I'm open to change here